### PR TITLE
Implemented TextFormat::colorize()

### PR DESCRIPTION
**Pull request by SOF3**
## Introduction

See 4c46087ffccc7f26835a1fc5bbf2ac60397884f9 for the initial discussion

### API changes

Added TextFormat::colorize().

### Behavioural changes

nil

## Backwards compatibility

No backward incompatibilities noted.

## Follow-up

nil

## Tests

```php
<?php
use pocketmine\utils\TextFormat;
include "vendor/autoload.php";
echo TextFormat::colorize("&a&b&c&d&e&f&g&h&i&j&&o& r&");
```

```
§a§b§c§d§e§f&g&h&i&j&§o& r&
```

### CODE REVIEW

**Comment by Thunder33345** [comment]
should a colourize for &colourname ex &green to be escape<greencode> considered while we are at this topic?
using if defined(TextFormat::class."::".$colourCode) and constant(TextFormat::class."::".$colourCode)

**Comment by dktapps** [comment]
too much work. This functionality matches (as far as I know) the Bukkit API and nothing more.

**Comment by Yexeed** [comment]
what means to much work?

**Comment by dktapps** [comment]
it means it's not worth the effort.

**Comment by SOF3** [comment]
@Thunder33345 In that case, an HTML-like approach e.g. `&green;` etc. (with a trailing semicolon) would be more appropriate. But in that case, you are inventing a new language.

File: src/pocketmine/utils/TextFormat.php [hunk 1]
```diff
@@ -80,6 +80,18 @@ public static function clean(string $string, bool $removeFormat = true) : string
...
+	/**
+	 * Replaces placeholders of § with the correct character. Only valid codes (as in the constants of the TextFormat class) will be converted.
+	 *
+	 * @param string $string
+	 * @param string $placeholder default "&"
+	 *
+	 * @return string
+	 */
+	public static function colorize(string $string, string $placeholder = "&") : string{
+		return preg_replace('/' . preg_quote($placeholder) . '([0-9a-fk-or])/u', TextFormat::ESCAPE . '$1', $string);
```
Comment (dktapps) [inline src/pocketmine/utils/TextFormat.php]: at least this regex is better than https://github.com/pmmp/PocketMine-MP/pull/1837/files#diff-b1b8c79c93533368678816cecafcbe98R65

**Comment by dktapps** [comment]
This will crash if `/` is used as a placeholder. Is this considered a problem?

**Review by dktapps — APPROVED** [review]

**Pull request merged**